src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -67,7 +69,7 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
                 _logger.Verbose($"Token cache could not satisfy silent request.");
 
-                if (AuthenticationRequestParameters.AppConfig.IsBrokerEnabled && ShouldTryWithBrokerError(ex.ErrorCode))
+                if (isBrokerConfigured && ShouldTryWithBrokerError(ex.ErrorCode))
```

Comment:
I would have preferred a solution where we don't have to expose this flag. For example, the iosBroker object can be made to respond to the AcquireTokenSilent by throwing an exception.

But anyway, this seems to have been tested, so it will have to do.